File: function_overloading.cpp
#include <bits/stdc++.h>
using namespace std;

class Calculator{
    public:
    int sum(int a, int b){
       return a+b;
    }
    int sum(int a, int b, int c){
        return a+b+c;
    }
};

int main(){
    Calculator c1;
    cout << c1.sum(1,2) << endl;
    cout << c1.sum(1,2,3) << endl;
}

File: function_overloading.py
class Calculator:
    def sum(self,n1=None,n2=None,n3=None):
        if n3:
            return n1+n2+n3
        else:
            return n1+ n2
        

c1 = Calculator()

print(c1.sum(1,2,3))
print(c1.sum(1,2))
